3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js

```diff
@@ -20,12 +20,21 @@ const rollDice = require('../../helpers/pokerDiceRoller');
 function rollTheDices() {
   // TODO Refactor this function
   const dices = [1, 2, 3, 4, 5];
-  return rollDice(1);
+  const resultPromises = [];
+  dices.forEach((dice) => {
+    resultPromises.push(rollDice(dice));
+  });
+  return resultPromises;
 }
 
-rollTheDices()
+Promise.all(rollTheDices())
   .then((results) => console.log('Resolved!', results))
   .catch((error) => console.log('Rejected!', error.message));
 
+//1. Dices that have not yet finished their roll continue to do so because  reject() doesn't mean 'return'.
+//2. There only a single call to either .then() or .catch() despite resolve() and/or reject() being called more than once
+//because .then() or .catch() can be called only ones. But I'm not sure :)
```

Comment:
If a promise becomes settled it must have either been resolved or rejected. It can only be resolved or rejected once. Therefore `.then()` or `.catch()` is called only once here. Additional calls to `resolve()` and `reject()` have no effect after that any more.